spirv_msl.cpp

```diff
@@ -3668,20 +3675,20 @@ string CompilerMSL::argument_decl(const SPIRFunction::Parameter &arg)
 	{
 		// Arrays of images and samplers are special cased.
 		decl += " (&";
-		decl += to_expression(name_id);
+		decl += builtin ? CompilerGLSL::to_expression(name_id) : to_expression(name_id);
```

Comment:
Shouldn't this just use CompilerGLSL::to_expression always? If it's not a builtin, the implementations will be equivalent anyways.